RestaurantReservationSystem.API/Controllers/EmployeesController.cs

```diff
@@ -0,0 +1,182 @@
+﻿using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.Mvc;
+using RestaurantReservationSystem.API.DTOs.Requests;
+using RestaurantReservationSystem.API.DTOs.Responses;
+using RestaurantReservationSystem.API.Responses;
+using RestaurantReservationSystem.API.Services.Interfaces;
+
+namespace RestaurantReservationSystem.API.Controllers
+{
+    /// <summary>
+    /// Controller for managing employee-related operations.
+    /// </summary>
+    [ApiController]
+    [Route("api/employees")]
+    public class EmployeesController : ControllerBase
+    {
+        private readonly IEmployeeService _employeeService;
+
+        public EmployeesController(IEmployeeService employeSservice)
+        {
+            _employeeService = employeSservice;
+        }
+
+        /// <summary>
+        /// Retrieves all managers.
+        /// </summary>
+        /// <returns>List of all managers.</returns>
+        [HttpGet("managers")]
```

Comment:
we can combine this api with the `GetAll` api. Update `GetAll` to have an optional query param to return managers only, if that's not passed, then you just return all.